From 5f21c4fa98f05423552da8716c80abfb7ca43091 Mon Sep 17 00:00:00 2001
From: Tushnim Bhattacharyya <tushnimb@codeaurora.org>
Date: Mon, 30 Apr 2018 17:25:38 -0700
Subject: qcacld-3.0: Add range check for mode param in cds_get_pcl

Add range check for mode parameter in cds_get_pcl.

Change-Id: I049bebc5ec62f925e164517d59f94d122dbc5dc2
CRs-Fixed: 2229615
---
 core/cds/src/cds_concurrency.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/cds/src/cds_concurrency.c b/core/cds/src/cds_concurrency.c
index 4a0fd1a..18823f7 100644
--- a/core/cds/src/cds_concurrency.c
+++ b/core/cds/src/cds_concurrency.c
@@ -5510,6 +5510,11 @@ QDF_STATUS cds_get_pcl(enum cds_con_mode mode,
 		return status;
 	}
 
+	if (mode >= CDS_MAX_NUM_OF_MODE) {
+		cds_err("requested mode:%d is not supported", mode);
+		return status;
+	}
+
 	/* find the current connection state from conc_connection_list*/
 	num_connections = cds_get_connection_count();
 
-- 
cgit v1.1

